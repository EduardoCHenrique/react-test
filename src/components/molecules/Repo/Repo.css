@import '../../../style/index.css';

.repo {
  border-bottom: 1px solid var(--border-color);
  padding: 2rem 0;
}

.repo__name {
  font-size: 2rem;
  font-weight: 400;
  margin-top: 0;
  color: var(--link-color);
}

.repo__name:hover {
  text-decoration: underline;
}

.repo__description {
  font-size: 1.4rem;
  font-weight: 400;
  color: var(--secondary-text-color);
}

.repo__footer {
  font-size: 1.2rem;
  font-weight: 400;
  color: var(--secondary-text-color);
}

.repo__footer__item {
  margin-right: 1rem;
}

.repo__footer__icon {
  font-size: 1.3rem;
}
