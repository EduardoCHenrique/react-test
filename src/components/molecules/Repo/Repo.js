import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { GoRepoForked, GoStar } from "react-icons/go";
import {Link} from 'react-router-dom';

import './Repo.css';

const Repo = ({className, id, name, description, language, stargazers_count, forks_count}) => {
  return(
    <div className={classNames('repo', className)}>
      <p className='repo__name'>
        <Link to={`/repos/${name}`}>{name}</Link>
      </p>
      <p className='repo__description'>{description}</p>

      <footer className='repo__footer'>
        <span className='repo__footer__item'>
          {language}
        </span>
        <span className="repo__footer__item">
          <GoStar className='repo__footer__icon'/>
          {stargazers_count}
        </span>

        <span className="repo__footer__item">
          <GoRepoForked className='repo__footer__icon'/>
          {forks_count}
        </span>
      </footer>
    </div>
  )
}

Repo.propTypes = {
  className: PropTypes.string,
  id: PropTypes.number,
  name: PropTypes.string,
  description: PropTypes.string,
  language: PropTypes.string,
  stargazers_count: PropTypes.number,
  forks_count: PropTypes.number
}

export default Repo