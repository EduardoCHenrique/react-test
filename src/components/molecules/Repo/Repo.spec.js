import React from 'react';
import { shallow } from 'enzyme';
import Repo from './Repo.js';

const default_props = {
  className: '',
  id: '',
  name: '',
  description: '',
  language: '',
  stargazers_count: 10,
  forks_count: 10,
};

function setup(props = default_props) {
  const wrapper = shallow(<Repo />);
  return { wrapper, props };
}

describe('Repo Test Suite', () => {
  it('Should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});