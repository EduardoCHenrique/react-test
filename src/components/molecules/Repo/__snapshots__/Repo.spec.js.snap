// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo Test Suite Should match snapshot 1`] = `ShallowWrapper {}`;
